src/codeql.ts

```diff
@@ -166,6 +173,29 @@ export async function downloadDatabase(
   }
 }
 
+export interface QueryMetadata {
+  kind?: string;
+}
+
+// Calls `resolve metadata` for the given query file and returns JSON output
+async function getQueryMetadata(
+  codeql: string,
+  query: string
+): Promise<QueryMetadata> {
+  const queryMetadata = await getExecOutput(codeql, [
+    "resolve",
+    "metadata",
+    "--format=json",
+    query,
+  ]);
+  if (queryMetadata.exitCode !== 0) {
+    throw new Error(
+      `Unable to run codeql resolve metadata. Exit code: ${queryMetadata.exitCode}`
+    );
+  }
+  return JSON.parse(queryMetadata.stdout) as QueryMetadata;
```

Comment:
Note: Some of the functions that read JSON output use the validator, e.g. 
```
  return validateObject(
    JSON.parse(bqrsInfoOutput.stdout, camelize),
    "bqrsInfo"
  );
```
My `as QueryMetadata` is bit annoying, but I didn't think it was helpful to create a JSON schema for the query metadata, since it might not exist and all the properties are optional 🙃

Reply:
Made a start here: https://github.com/github/codeql-variant-analysis-action/pull/849. All looks good, except that failing test I mentioned above 🧪 😅